Client/Frontend/Autofill/CreditCard/ViewComponents/CreditCardInput/CreditCardInputViewModel.swift

```diff
@@ -0,0 +1,95 @@
+        self.fieldType = fieldType
+
+        switch fieldType {
+        case .name:
+            self.fieldHeadline = .CreditCard.EditCard.NameOnCardTitle
+            self.errorString = .CreditCard.ErrorState.NameOnCardSublabel
+            self.delimiterCharacter = nil
+            self.formattedTextLimit = 100 // TBA
+            self.keyboardType = .alphabet
+        case .number:
+            self.fieldHeadline = .CreditCard.EditCard.CardNumberTitle
+            self.errorString = .CreditCard.ErrorState.CardNumberSublabel
+            self.formattedTextLimit = 20
+            self.delimiterCharacter = " "
+            self.keyboardType = .numberPad
+        case .expiration:
+            self.fieldHeadline = .CreditCard.EditCard.CardExpirationDateTitle
+            self.errorString = .CreditCard.ErrorState.CardExpirationDateSublabel
+            self.formattedTextLimit = 7
+            self.delimiterCharacter = " / "
+            self.keyboardType = .numberPad
+        case .securityCode: // Design feedback needed
+            self.fieldHeadline = "Security Code"
+            self.errorString = "Enter the three digit security code"
+            self.formattedTextLimit = 3
+            self.delimiterCharacter = nil
+            self.keyboardType = .numberPad
+        }
+    }
+
+    func updateNameValidity(userInputtedText: String) {
+        isValid = !userInputtedText.isEmpty
+    }
+
+    func updateCardNumberValidity(userInputtedText: String) {
+        isValid = userInputtedText.count >= 15 && userInputtedText.count <= 16
+    }
+
+    func updateExpirationValidity(userInputtedText: String) {
+        let year = Calendar(identifier: .gregorian).dateComponents([.year], from: Date()).year ?? 2023
+
+        guard let month = Int(userInputtedText.prefix(2)),
+              let inputYear = Int(userInputtedText.suffix(2)),
+              userInputtedText.count < 5,
+              month < 13 && month > 0,
+              (inputYear + 2000) < 2099 && (inputYear + 2000) >= year else {
+            isValid = false
+            return
+        }
+
+        isValid = true
+    }
+
+    func updateSecurityCodeValidity(userInputtedText: String) {
+        isValid = userInputtedText.count == 3
+    }
+
+    /// This function takes a credit card input string and returns it in a user readable format.
+    /// - Parameters:
+    ///   - inputType: The `CreditCardInputType`.
+    ///   - textInput: The user inputted string.
+    /// - Returns: The string in the expected and readable format for that `inputType`.
+    func separate(inputType: CreditCardInputType, for textInput: String) -> String? {
+        guard let delimiterCharacter = delimiterCharacter, textInput.count <= formattedTextLimit else { return nil }
+
+        var formattedText = ""
+        switch inputType {
+        case .number:
+            formattedText = textInput.enumerated().map { $0.isMultiple(of: 4) && ($0 != 0) ? "\(delimiterCharacter)\($1)" : String($1) }.joined()
```

Comment:
This line (and line 88) are way too long (more than 160 characters). I'd suggest moving the closure on the next line.